src/cpp/commandlist.cpp

```diff
@@ -30,63 +30,76 @@
 
 using namespace SmartRedis;
 
+// CommandList copy constructor
 CommandList::CommandList(const CommandList& cmd_lst)
 {
     std::vector<Command*>::const_iterator c_it = cmd_lst._commands.cbegin();
-    std::vector<Command*>::const_iterator c_it_end = cmd_lst._commands.cend();
-    Command* new_cmd;
-    for(; c_it != c_it_end; c_it++) {
-        new_cmd = (*c_it)->clone();
+    for ( ; c_it != cmd_lst._commands.cend(); c_it++) {
+        Command* new_cmd = (*c_it)->clone();
+        if (new_cmd == NULL) {
```

Comment:
I don't think we need this check.  It doesn't seem like the ``clone()`` functions could ever return a NULL pointer (and we should enforce that they don't because we have full control of the API).

Reply:
I disagree strongly on this one. Second rule of defensive programming is to assume that any other routine you call has the potential to give back a bad result -- whether through a bad input, a programming error, or an environmental issue of some sort. All return values from other functions must be sanity checked.